### PR TITLE
Allow running a portion of binaryen test suite

### PR DESCRIPTION
**Pull request by quantum5**

### CODE REVIEW

**Comment by tlively** [comment]
Can you add some usage examples in a comment at the top of check.py?

File: check.py [hunk 1]
```diff
@@ -522,30 +523,32 @@ def run_unittest():
...
+  ('dylink', run_dylink_tests),
+  ('ctor-eval', run_ctor_eval_tests),
+  ('wasm-metadce', run_wasm_metadce_tests),
+  ('wasm-reduce', lambda: run_wasm_reduce_tests() if has_shell_timeout() else None),
+  ('spec', run_spec_tests),
+  ('binaryenjs', binaryenjs.test_binaryen_js),
+  ('lld', lld.test_wasm_emscripten_finalize),
+  ('wasm2js', wasm2js.test_wasm2js),
+  ('validator', run_validator_tests),
+  ('vanilla', lambda: run_vanilla_tests() if has_vanilla_emcc and has_vanilla_llvm and 0 else None),
```
Comment (sbc100) [inline check.py]: Would it make sense to move these predicates into the functions themselves as early returns?

**Comment by aheejin** [comment]
> Can you add some usage examples in a comment at the top of check.py?

How about having a proper help message about the usage for `-h` or something?

File: check.py [hunk 2]
```diff
@@ -522,30 +523,32 @@ def run_unittest():
     raise Exception("unittest failed")
 
 
+TEST_SUITES = OrderedDict([
+  ('help', run_help_tests),
```
Comment (jgravelle-google) [inline check.py]: If I'm reading this correctly, we'd say `check.py help` at the command line, and that would run the help tests. As opposed to `check.py --help` which would print usage information for check.py.

How about `'help-messages'` (or `'test-help-messages'`) to disambiguate?

Comment (quantum5) [reply check.py]: Done!

File: check.py (hunk 1 above)
Comment (quantum5) [reply check.py]: Done!

**Pull request merged**